pkg/flist/flist.go

```diff
@@ -197,13 +202,22 @@ func (f *flistModule) mount(name, url, storage string, opts pkg.MountOptions) (s
 				return "", fmt.Errorf("invalid mount option, missing disk type and/or size")
 			}
 
-			path, err = f.storage.CreateFilesystem(name, opts.Limit*mib, opts.Type)
+			backend, err = f.storage.CreateFilesystem(name, opts.Limit*mib, opts.Type)
 			if err != nil {
 				return "", errors.Wrap(err, "failed to create read-write subvolume for 0-fs")
 			}
 		}
+	}
 
-		args = append(args, "-backend", path)
+	if len(backend) != 0 {
+		args = append(args, "-backend", backend)
+		// in case of an error (mount is never fully completed)
+		// we need to deallocate the filesystem
+		defer func() {
+			if err != nil {
+				f.storage.ReleaseFilesystem(name)
```

Comment:
You still cannot do that here. Cause it might be you are trying to mount an flist after a reboot for which there is actual data on the backend. So even here I don't think we can release the filesystem.

Reply:
I think with some code reorganization we can grantee this is a new allocation that was never get used. There is already a check that this is already a valid mount, that I think should be moved a bit higher.

Reply:
yep let's do that.